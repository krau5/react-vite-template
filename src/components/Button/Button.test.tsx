import { renderComponent, user } from 'testUtils';
import { Button } from '.';

it('renders a component', async () => {
  const handleClick = jest.fn(() => {});

  const { asFragment, findByText } = renderComponent(
    <Button onClick={handleClick}>Caption</Button>,
  );

  expect(asFragment()).toMatchSnapshot();

  const button = await findByText('Caption');

  await user.click(button);

  expect(handleClick).toHaveBeenCalled();
  expect(handleClick).toHaveBeenCalledTimes(1);
});
