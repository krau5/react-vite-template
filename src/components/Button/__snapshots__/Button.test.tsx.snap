// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a component 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding: 8px 16px;
  border: none;
  outline: none;
  cursor: pointer;
  border-radius: 8px;
  background: #0088cc;
  color: #ffffff;
}

.emotion-0:hover {
  background: #0197e2;
}

.emotion-0:active {
  background: #0082c2;
}

<button
    class="emotion-0"
    type="button"
  >
    Caption
  </button>
</DocumentFragment>
`;
